app/v2/default.go

```diff
@@ -0,0 +1,72 @@
+package app
+
+import (
+	"context"
+
+	fkitlog "github.com/arquivei/foundationkit/log"
+
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	// This is the default app.
+	defaultApp *App
+)
+
+type AppConfig interface {
+	GetAppConfig() Config
+}
+
+// Bootstrap initializes the config structure, the log and creates a new app internally.
+func Bootstrap(appVersion string, config AppConfig) {
+	SetupConfig(config)
+	appConfig := config.GetAppConfig()
+	fkitlog.SetupLogger(appConfig.App.Log, appVersion)
+
+	log.Info().Str("config", fkitlog.Flatten(config)).Msg("[app] Configuration loaded and global logger configured.")
+	defaultApp = New(appConfig)
+}
+
+// RunAndWait executes the main loop on a go-routine and listens to SIGINT and SIGKILL to start the shutdown.
+// This is expected to be called only once and will panic if called a second time.
```

Comment:
Nit: maybe the documentation could  be among the lines of "RunAndWait is a passthrough to the default App RunAndWait"? This would avoid the description of "RunAndWait" being in multiple places, and also avoid the risk of one being updated but not the other when the code changes.